### PR TITLE
Fix `Delayed` optimization issue 

### PR DESCRIPTION
**Pull request by jrbourbeau**
Closes https://github.com/dask/dask-ml/issues/877
Closes https://github.com/dask/dask/issues/8360

### CODE REVIEW

File: dask_ml/_partial.py [hunk 1]
```diff
@@ -120,19 +121,10 @@ def fit(
...
 
-    graphs = {x_name: x.__dask_graph__(), name: dsk}
-    if hasattr(y, "__dask_graph__"):
-        graphs[y_name] = y.__dask_graph__()
-
-    try:
-        from dask.highlevelgraph import HighLevelGraph
-
-        new_dsk = HighLevelGraph.merge(*graphs.values())
-    except ImportError:
```
Comment (jrbourbeau) [inline dask_ml/_partial.py:131]: Note: this `ImportError` branch is no longer needed based on our current lower bound of supported `dask` versions

https://github.com/dask/dask-ml/blob/f250432bba9046cdca3b464cb6e8eede879d3284/setup.py#L13-L14

**Comment by jrbourbeau** [comment]
Planning to merge later today if there are no further comments

**Comment by VibhuJawa** [comment]
Thanks for working on this @jrbourbeau  and unblocking the CI .

**Comment by jakirkham** [comment]
Thanks James! 😄

**Pull request merged**